xdsl/irdl/declarative_assembly_format.py

```diff
@@ -708,27 +722,51 @@ class AttributeVariable(FormatDirective):
     The directive will request a space to be printed right after.
     """
 
-    attr_name: str
+    name: str
     """The attribute name as it should be in the attribute or property dictionary."""
     is_property: bool
     """Should this attribute be put in the attribute or property dictionary."""
+    unique_base: type[Attribute] | None
```

Comment:
Instead of this being the unique base, even for builtin attributes, I would set this to `None` for builtin attributes.
I think that this design would make more sense, what do you think?
It will also solve later the `qualified` directive, where we will not want the unique base, even though it's known.

I would just change this and rename the docstring to
```
"""
The elided attribute base, if any.
When this value is set, the attribute name is not printed nor parsed.
For instance, `!test.attr<param>` would be parsed and printed as `<param>`.
"""
```